### PR TITLE
2499: Proposals to improve js execution time

### PR DESCRIPTION
**Pull request by sarahsporck**
### Short description

Proposal to reduce js execution time and minimize main thread work. Feel free to use this pr as inspiration/starting point for #2499 and #2498. Also feel free to close this PR if you don't see the changes working

### Proposed changes

- set `sideEffects: false` in `package.json` to allow tree shaking our packages
- use splitChunks plugin to split the code into react, dompurify (as it is quite big and cannot be tree-shaken), node_modules & our code. This can probably be optimized further
- get rid of lodash in production code (check out the bundle analyzer)

### Related issues

Relates: #2498 #2499 

---

### CODE REVIEW

File: web/src/components/PoiDetails.tsx [hunk 1]
```diff
@@ -136,7 +136,7 @@ const PoiDetails = ({ poi, distance, toolbar }: PoiDetailsProps): ReactElement =
   const { content, location, website, phoneNumber, email, isCurrentlyOpen, openingHours, temporarilyClosed, category } =
     poi
 
-  const thumbnail = poi.thumbnail?.replace('-150x150', '') ?? PoiThumbnailPlaceholderLarge
+  const thumbnail = poi.thumbnaila?.replace('-150x150', '') ?? PoiThumbnailPlaceholderLarge
```
Comment (f1sh1918) [inline web/src/components/PoiDetails.tsx:139]: typo?

File: web/tools/webpack.config.ts [hunk 2]
```diff
@@ -351,7 +382,7 @@ const createConfig = (
   // Optimize the bundle in production mode
   // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
   if (!devServer && optimize) {
-    config.plugins?.push(new optimize.AggressiveMergingPlugin())
+    config.plugins?.push(new optimize.AggressiveMergingPlugin(), new optimize.AggressiveSplittingPlugin({}))
```
Comment (f1sh1918) [inline web/tools/webpack.config.ts:385]: i get a deprecation warning
```
(node:14657) [DEP_WEBPACK_AGGRESSIVE_SPLITTING_PLUGIN] DeprecationWarning: AggressiveSplittingPlugin is deprecated in favor of SplitChunksPlugin
```

**Comment by f1sh1918** [comment]
Looks nice :)
Bundle sizes are super small! There is a lot improvement!
Lighthouse doesn't show me much performance improvements tbh, so. The largest contentful paint is still at ~8s but a little bit faster than on main

File: web/tools/webpack.config.ts (hunk 2 above)
Comment (sarahsporck) [reply web/tools/webpack.config.ts:385]: Argh. This is not even required, because I added split chunks above.

**Comment by steffenkleinle** [comment]
I opened #2776 without the webpack changes and a few further code optimizations and cleanup.

**Pull request closed without merging**